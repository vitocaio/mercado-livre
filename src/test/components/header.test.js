import React from 'react'
import renderer from 'react-test-renderer';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { BrowserRouter as Router } from "react-router-dom";

import Header from '../../components/header';

configure({ adapter: new Adapter() });

const renderHeader = () => {
  return shallow(
    <Router>
      <Header />
    </Router>
  ).shallow();
};

// Snapshot for Home React Component
describe('>>>HEADER --- Snapshot', () => {
  const wrapper = renderHeader();
  it('+++capturing Snapshot of Header', () => {
    const renderedValue = renderer.create(wrapper).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe('>>>HEADER --- Shallow Render REACT COMPONENTS', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = renderHeader()
  });

  it('+++ render the DUMB component', () => {
    expect(wrapper.length).toEqual(1);
  });
});

describe('>>>HEADER --- attributes', () => {
  const props = {
    onSearch: jest.fn(),
  };
  const header = mount(
    <Router>
      <Header {...props} />
    </Router>
  );

  it('+++ should have logo', () => {
    expect(header.find('img').props('src').src).toEqual('Logo_ML.png');
  });

  it('+++ should have input search', () => {
    expect(header.find('input').props('type').type).toEqual('search');
  });
});

