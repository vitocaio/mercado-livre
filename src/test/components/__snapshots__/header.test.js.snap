// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>HEADER --- Snapshot +++capturing Snapshot of Header 1`] = `
<header
  className="page-header"
>
  <div
    className="sc-bwzfXH kZwgyh"
  >
    <nav
      className="navbar navbar-expand-lg sc-htpNat eRFqyv"
    >
      <button
        aria-controls="navbarCollapse"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-target="#navbarCollapse"
        data-toggle="collapse"
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-bars fa-w-14 sc-bdVaJa fePDys"
          data-icon="bars"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
      <a
        className="navbar-brand sc-bxivhb cKPwfs"
        href="/"
      >
        <img
          alt="Logo ML"
          className="img-logo-ml-sm-white"
          src="Logo_ML.png"
          width="60px"
        />
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarCollapse"
      >
        <ul
          className="navbar-nav mr-auto"
        >
          <li>
            <form
              className="form-inline"
              onSubmit={[Function]}
            >
              <div
                className="sc-ifAKCX iCqXpP"
              >
                <button
                  className="btn"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-search fa-w-16 sc-bdVaJa fePDys"
                    data-icon="search"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </button>
                <input
                  aria-label="Search"
                  className="form-control"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Search"
                  type="search"
                  value=""
                />
              </div>
              <span
                className="sc-EHOje cMekiA"
              />
            </form>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</header>
`;
